From 281618d71a20ad7b5e6f2b6b06949608bb677939 Mon Sep 17 00:00:00 2001
From: Wujian sun <wujian.sun@nxp.com>
Date: Wed, 17 Oct 2018 14:33:26 +0800
Subject: [PATCH 4882/5242] MGS-4281 [#imx-1219] Importing an invalid dmabuf
 fd causes the kernel to oops

commit  00acbacf48cf034648a73ee8b54975c6a3a84fa2 from
https://source.codeaurora.org/external/imx/linux-imx.git

Error return for invalid dmabuf,if set invalid client_buffer.

Date: 17th Oct, 2018.
Signed-off-by Wujian Sun <wujian.sun@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../hal/kernel/gc_hal_kernel_video_memory.c        |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel_video_memory.c b/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel_video_memory.c
index 2eb6a65..033494b 100644
--- a/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel_video_memory.c
+++ b/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel_video_memory.c
@@ -3284,6 +3284,9 @@ static void _dmabuf_kunmap(struct dma_buf *dmabuf, unsigned long offset, void *p
             /* Import dma buf handle. */
             dmabuf = dma_buf_get(fd);
 
+            if (IS_ERR(dmabuf))
+                return PTR_ERR(dmabuf);
+
             Desc->handle = -1;
             Desc->dmabuf = gcmPTR_TO_UINT64(dmabuf);
 
-- 
1.7.9.5

